nntrainer/models/neuralnet.cpp

```diff
@@ -704,13 +713,18 @@ int NeuralNetwork::train_run() {
     stat.num_iterations++;
   };
 
-  auto train_epoch_end = [this](RunStats &stat, DataBuffer &buffer) {
+  auto train_epoch_end = [this, &train_log_file](RunStats &stat,
+                                                 DataBuffer &buffer) {
     stat.loss /= static_cast<float>(stat.num_iterations);
     auto &save_path = std::get<props::SavePath>(model_flex_props);
     if (!save_path.empty()) {
       save(save_path, ml::train::ModelFormat::MODEL_FORMAT_BIN);
     }
 
+    if (train_log_file.good()) {
```

Comment:
I think it's better to check if prop exists and try this while check if train_log_file is bad, we throw